ipet/parsing/ReaderManager.py

```diff
@@ -296,46 +302,71 @@ def collectData(self):
         """
         assert(self.testrun != None)
 
-        # sort the files by context
-        self.filestrings = sorted(self.filestrings, key = lambda x:self.sortingKeyContext(self.filenameGetContext(x)))
+        # sort the files by context, for example: outfiles should be read before solufiles
+        if not self.testrun.inputfromstdin:
```

Comment:
Reading something from a filename should soon be deprecated IMHO. All of this, especially the settings, are what I consider Metainformation. Meta information is not relevant in itself, but for combining runs in an experiment, it might be relevant. My idea with Meta-Information is the following:

It can either be specified in an outfile directly as follows:

\@Settings default
\@Seed 5
\@Machine opt55


Or in a file with the same basename and the .meta extensions as follows, e.g., check.[blablabla].meta:
"""
Settings default
Seed 5
Machine opt55
"""
This, however, is not part of this review.
or, alternatively, in the outfile directly, as follows: